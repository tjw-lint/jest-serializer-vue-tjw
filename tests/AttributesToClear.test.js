const helpers = require('./helpers.js');

import { shallowMount } from '@vue/test-utils';
import AttributesToClear from './components/AttributesToClear.vue';

describe('AttributesToClear.vue', () => {
  test('Default snapshot', () => {
    helpers.mockSettings({ removeDataTest: false });

    const wrapper = shallowMount(AttributesToClear);

    expect(wrapper)
      .toMatchSnapshot();
  });

  test('Just ID cleared', () => {
    helpers.mockSettings({
      removeDataTest: false,
      attributesToClear: ['id']
    });

    const wrapper = shallowMount(AttributesToClear);

    expect(wrapper)
      .toMatchSnapshot();
  });

  test('All attributes cleared', () => {
    helpers.mockSettings({
      removeDataTest: false,
      attributesToClear: [
        'id',
        'class',
        'title',
        'asdf',
        'data-test ',
        ' some junk ',
        true
      ]
    });

    const wrapper = shallowMount(AttributesToClear);

    expect(wrapper)
      .toMatchSnapshot();
  });
});
