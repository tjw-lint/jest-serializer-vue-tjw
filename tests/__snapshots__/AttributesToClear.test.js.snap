// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttributesToClear.vue All attributes cleared 1`] = `
<div>
  <h1 id title class>
    fluffy
  </h1>
  <h2 id asdf>
    mittens
  </h2>
  <h3 id asdf data-test>
    scruffy
  </h3>
</div>
`;

exports[`AttributesToClear.vue Default snapshot 1`] = `
<div>
  <h1 id="kitten" title="cat" class="meow nyan">
    fluffy
  </h1>
  <h2 id="kitten2" asdf="qwer">
    mittens
  </h2>
  <h3 id asdf data-test="qwer">
    scruffy
  </h3>
</div>
`;

exports[`AttributesToClear.vue Just ID cleared 1`] = `
<div>
  <h1 id title="cat" class="meow nyan">
    fluffy
  </h1>
  <h2 id asdf="qwer">
    mittens
  </h2>
  <h3 id asdf data-test="qwer">
    scruffy
  </h3>
</div>
`;
